media-library/overview/upload-files.md

```diff
@@ -52,3 +52,135 @@ You can share a folder with a restricted user by providing "Can contribute" or "
+
+#### Overwrite user-set tags during upload
+
+If this feature is enabled and a file with the same name already exists at the target destination, the user-set tags of the file will be replaced with new ones or removed if no tags are provided. Conversely, if this setting is off, the existing user-set tags of the file will be preserved in the new version.
+
+#### Overwrite custom metadata during the upload
+
+If this feature is enabled and a file with the same name already exists at the target destination, the custom metadata of the file will be replaced with a new one or removed if no custom metadata is provided. Conversely, if this setting is off, the existing custom metadata of the file will be preserved in the new version.
+
+#### Pre transformation
+
+Pre-transformation lets you modify image or video files before they're uploaded. You can only request a single pre-transformation on an asset. This feature is applicable exclusively to image and video files and any valid supported [image transformation](../../features/image-transformations/) or [video transformation](../../features/video-transformation/) can be applied to their respective files.
+
+<figure><img src="../../.gitbook/assets/pre-transformation.png" alt=""><figcaption><p>Pre transformation</p></figcaption></figure>
+
+#### Post transformation
+
+Post-transformations help you eagerly generate transformations after an image or video file has been uploaded to ensure fast delivery to your users. This feature is applicable exclusively to image and video files. You can apply several post-transformations to an asset, with a maximum limit of five allowed post-transformations.&#x20;
+
+<figure><img src="../../.gitbook/assets/post-transformation.png" alt=""><figcaption><p>Post transformation</p></figcaption></figure>
+
+#### Custom metadata
+
+Custom metadata fields can be used to store any type of data about your images in key-value pairs during upload. Before setting any custom metadata on an asset you have to create the field from media library settings or using custom metadata fields API.
+
+#### Tags
+
+Imagekit enables you to include tags during the upload of assets, this setting is not available in the default upload parameters within the media library settings. Tags can be chosen from an auto-completer or created as new ones.
+
+<figure><img src="../../.gitbook/assets/tag-during-upload.png" alt=""><figcaption><p>Tag</p></figcaption></figure>
+
+#### Webhook URL
+
+You can include a `webhookUrl` parameter during file upload. The final status of extensions after they have completed execution will be delivered to this endpoint as a POST request. This setting is not available in the default upload parameters within the media library settings.  For more information about webhookUrl, refer to this [resource](../../api-reference/api-introduction/webhooks.md).
```

Comment:
Does this also work for pre/post transforms?

Reply:
no